codec-http/src/main/java/io/netty/handler/codec/http/HttpDecoderConfig.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2023 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+import static io.netty.util.internal.ObjectUtil.checkPositive;
+
+/**
+ * A configuration object for specifying the behaviour of {@link HttpObjectDecoder} and its subclasses.
+ * <p>
+ * The {@link HttpDecoderConfig} objects are mutable to reduce allocation,
+ * but also {@link Cloneable} in case a defensive copy is needed.
+ */
+public final class HttpDecoderConfig implements Cloneable {
```

Comment:
IIRC, there was a recommendation to use a copy-ctor pattern vs implementing a `Clineable` interface somewhere, maybe in Effective Java?

Reply:
What was the argument for that?

Reply:
I don't know if [this article from 2000](https://adtmag.com/articles/2000/01/18/effective-javaeffective-cloning.aspx) is the latest word on that, but since the field types are not expected to have interior mutability, I think it's fine here.

Reply:
Yes, that's the one. 
It's like a question when to use Atomic* vs Atomic*Updater. Sometimes it's easier to stick to a recommended pattern instead of considering each case independently. Up to you what approach you would like to keep.